common/achievement/util.ts

```diff
@@ -1,28 +1,56 @@
-import { ActionID, NotificationContext } from '../notification/types';
+import { ActionID } from '../notification/types';
 import { metricsByAction } from './metrics';
-import { Metric } from './types';
-
-export function getRelationByContext(context: NotificationContext): string {
-    const { appointment, match, subcourse } = context;
-    let relation: string;
-
-    switch (context) {
-        case context.match:
-            relation = `match/${match.id}`;
-            break;
-        case context.subcourse:
-            relation = `subcourse/${subcourse.id}`;
-            break;
-        case context.appointment:
-            relation = `appointment/${appointment.id}`;
-            break;
-        default:
-            relation = '';
+import { Metric, RelationContextType } from './types';
+import { prisma } from '../prisma';
+import { getLogger } from '../logger/logger';
+
+const logger = getLogger('Gamification');
+export function isGamificationFeatureActive(): boolean {
+    const isActive: boolean = JSON.parse(process.env.GAMIFICATION_ACTIVE || 'false');
+
+    if (!isActive) {
+        logger.warn('Gamification is deactivated');
     }
 
-    return relation;
+    return isActive;
+}
+
+export function assureGamificationFeatureActive() {
+    if (!isGamificationFeatureActive()) {
+        return false;
+    }
+    return true;
 }
 
 export function getMetricsByAction<ID extends ActionID>(actionId: ID): Metric[] {
     return metricsByAction.get(actionId) || [];
 }
+
+type RelationTypes = 'match' | 'subcourse';
+
+export function getRelationTypeAndId(relation: string): [relationType: RelationTypes, id: number] {
+    const validRelationTypes = ['match', 'subcourse'];
+    const [relationType, relationId] = relation.split('/');
+    if (!validRelationTypes.includes(relationType)) {
+        throw Error('No valid relation found in relation: ' + relationType);
+    }
```

Comment:
the relations are stored as strings so considering the db model this would not be possible. I actually do not know why this is the case, but there must be a very good reason.